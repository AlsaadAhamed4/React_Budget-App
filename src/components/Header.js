import { Link } from 'react-router-dom'
import React from 'react';
import { startLogOut } from '../Actions/auth';
import { connect } from 'react-redux';

export const Header = ({ startLogOut }) => (
    <header className='header'>
        <div className='content-container'>
            <div className='header__content'>
                <Link className='header__link' exact="true" to='/dashboard'><h1>Budget-App</h1></Link>
                <button className='button button--link ' onClick={startLogOut}>Logout</button>
            </div>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogOut: () => dispatch(startLogOut())
});

export default connect(undefined, mapDispatchToProps)(Header);