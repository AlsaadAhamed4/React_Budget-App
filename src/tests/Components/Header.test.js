//virtually rendering the component
//react -library to do it react-test-render

//import ReactShallowRenderer from 'react-test-renderer/shallow';
import React from 'react'
import Header from '../../components/Header';
import { shallow } from 'enzyme'


test('should render header correctly', () => {
    const wrapper = shallow(<Header/>);
    expect(wrapper).toMatchSnapshot();
    // const renderer = new ReactShallowRenderer
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
    //console.log(renderer.getRenderOutput());
})
