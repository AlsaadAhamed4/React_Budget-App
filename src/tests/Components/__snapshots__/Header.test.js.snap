// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <[object Object]
        className="header__link"
        exact={true}
        to="/dashboard"
      >
        <h1>
          Budget-App
        </h1>
      </[object Object]>
      <button
        className="button button--link "
        onClick={[MockFunction]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
